crates/turborepo-lib/src/framework.rs

```diff
@@ -0,0 +1,290 @@
+                },
+            },
+            Framework {
+                slug: "solidstart",
+                env_wildcards: vec!["VITE_*"],
+                dependency_match: Matcher {
+                    strategy: Strategy::All,
+                    dependencies: vec!["solid-js", "solid-start"],
+                },
+            },
+            Framework {
+                slug: "vue",
+                env_wildcards: vec!["VUE_APP_*"],
+                dependency_match: Matcher {
+                    strategy: Strategy::All,
+                    dependencies: vec!["@vue/cli-service"],
+                },
+            },
+            Framework {
+                slug: "sveltekit",
+                env_wildcards: vec!["VITE_*"],
+                dependency_match: Matcher {
+                    strategy: Strategy::All,
+                    dependencies: vec!["@sveltejs/kit"],
+                },
+            },
+            Framework {
+                slug: "create-react-app",
+                env_wildcards: vec!["REACT_APP_*"],
+                dependency_match: Matcher {
+                    strategy: Strategy::Some,
+                    dependencies: vec!["react-scripts", "react-dev-utils"],
+                },
+            },
+            Framework {
+                slug: "nuxtjs",
+                env_wildcards: vec!["NUXT_ENV_*"],
+                dependency_match: Matcher {
+                    strategy: Strategy::Some,
+                    dependencies: vec!["nuxt", "nuxt-edge", "nuxt3", "nuxt3-edge"],
+                },
+            },
+            Framework {
+                slug: "redwoodjs",
+                env_wildcards: vec!["REDWOOD_ENV_*"],
+                dependency_match: Matcher {
+                    strategy: Strategy::All,
+                    dependencies: vec!["@redwoodjs/core"],
+                },
+            },
+            Framework {
+                slug: "vite",
+                env_wildcards: vec!["VITE_*"],
+                dependency_match: Matcher {
+                    strategy: Strategy::All,
+                    dependencies: vec!["vite"],
+                },
+            },
+            Framework {
+                slug: "sanity",
+                env_wildcards: vec!["SANITY_STUDIO_*"],
+                dependency_match: Matcher {
+                    strategy: Strategy::All,
+                    dependencies: vec!["@sanity/cli"],
+                },
+            },
+        ]
+    })
+}
+
+impl Matcher {
+    pub fn test(&self, workspace: &WorkspaceInfo, is_monorepo: bool) -> bool {
+        // In the case where we're not in a monorepo, i.e. single package mode
+        // `unresolved_external_dependencies` is not populated. In which
+        // case we should check `dependencies` instead.
+        let deps = if is_monorepo {
+            workspace.unresolved_external_dependencies.as_ref()
+        } else {
+            workspace.package_json.dependencies.as_ref()
```

Comment:
I realize this is just a port of Go behavior, but do we want to also check the other places dependencies could appear? Primarily `devDependencies` and `optionalDependencies`